docs/Technical-Documentation/tech-memos/login-gov-set-integration/login-gov-set-integration.md

```diff
@@ -0,0 +1,358 @@
+# Login.gov Security Event Token (SET) Integration
+
+**Audience**: TDP Software Engineers <br>
+**Subject**: Integration of Login.gov Security Event Tokens with Account Recreation Handling <br>
+**Date**: July 17, 2025 <br>
+
+## Summary
+This technical memorandum outlines the implementation plan for integrating Security Event Tokens (SETs) from Login.gov into the TDP application. The integration will enable TDP to:
+1. Handle security-related notifications about user accounts
+2. Automatically manage account recreation scenarios when users delete and recreate their Login.gov accounts
+3. Maintain audit history and respond appropriately to security events
+
+## Background
+Login.gov provides a Security Event Token (SET) notification system that follows the OpenID RISC Event Types specification. These notifications are sent as JWT tokens via HTTP POST requests when specific security events occur for users, such as:
+- Account disabling
+- Account purging (when users delete their accounts)
+- Password resets
+- Recovery information changes
+
+A common issue in production is that users sometimes delete their Login.gov accounts and recreate them, which results in a new `sub` claim for the same email address. This causes authentication issues in TDP since it relies on the `sub` claim for user identification. By implementing SET handling, particularly for the "Account Purged" event, we can automatically manage these account recreation scenarios without requiring additional user verification steps.
+
+## Out of Scope
+* Changes to the existing Login.gov authentication flow
+* Frontend UI components for displaying security events
+* Integration with other identity providers' security event systems
+* Real-time notifications to users about security events
+
+## Method/Design
+
+### Security Event Token Model
+
+Create a model to store Security Event Tokens from Login.gov.
```

Comment:
Do we need to store these events? ~~And are there events that we don't handle that might just end up as noise?~~

Reply:
'handled' my own question about events